pkg/handlers/deploy.go

```diff
@@ -93,6 +102,23 @@ func (h *Handler) DeployAppVersion(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	kotsUpgradeNeeded, targetVersion, err := isKotsUpgradeNeeded(a, int64(sequence))
```

Comment:
the previous if statement already detects of the version is pending download or not based on compatibility. will this code ever be reached?